### PR TITLE
Fix contract originated address

### PR DESCRIPTION
**Pull request by renatoalencar**
## Problem

Contract origination is using the  hash of the `User_operation`, which is the same for the same `sender * code * storage` triple. The would make possible to overwrite a contract's storage with it's initial value, also making not possible for an implicit account to have multiple contracts with same initial state.

## Solution

Use the protocol operation hash instead, which depends on stuff like block height and nonce. Also adds an endpoint for making possible to test it.

### CODE REVIEW

File: src/bin/deku_node.ml [hunk 1]
```diff
@@ -158,6 +158,23 @@ let handle_ticket_balance =
       let amount = Flows.request_ticket_balance state ~ticket ~address in
       Ok { amount })
 
+(* POST /get-storage *)
```
Comment (d4hines) [inline src/bin/deku_node.ml:161]: This additional endpoint seems unrelated to the goal of the PR. I also don't think we want to expose Core_deku state via the Deku node since this conflicts with the architecture design of Deku p (potentially). See #535.

**Review by d4hines — CHANGES_REQUESTED** [review]

File: src/core_deku/state.ml [hunk 2]
```diff
@@ -97,9 +97,9 @@ let apply_user_operation t user_operation =
...
 
-let apply_user_operation t user_operation =
-  match apply_user_operation t user_operation with
+let apply_user_operation t hash user_operation =
+  match apply_user_operation t hash user_operation with
   | Ok (t, receipt) -> (t, receipt)
   (* TODO: use this erros for something *)
   | Error (`Origination_error _ | `Invocation_error _) -> (t, None)
-  | Error `Not_enough_funds -> (t, None)
+  | Error (`Insufficient_funds | `Not_enough_funds) -> (t, None)
```
Comment (d4hines) [inline src/core_deku/state.ml:105]: Looks like we added an extra error here? `Insufficient` is synonym to `Not_enough` so we ought not have both.

Comment (renatoalencar) [reply src/core_deku/state.ml]: Sorry, I was tricked by my local environment mentioning the `this pattern is not exhaustive error`. Fixed.

**Review by d4hines — APPROVED** [review]
LGTM
